src/Illuminate/Foundation/helpers.php

```diff
@@ -636,7 +636,7 @@ function report($exception)
      *
      * @param  array|string|null  $key
      * @param  mixed  $default
-     * @return mixed
+     * @return mixed|\Illuminate\Http\Request|string|array|null
```

Comment:
`mixed` can never be combined with something else.

Reply:
Look in the same file, you will see literally four `mixed` in combination with other types:

- [@return mixed|\Illuminate\Contracts\Foundation\Application](https://github.com/laravel/framework/blob/ab1506091b9f166b312b3990d07b2e21d971f2e6/src/Illuminate/Foundation/helpers.php#L111)
- [@return mixed|\Illuminate\Cache\CacheManager](https://github.com/laravel/framework/blob/ab1506091b9f166b312b3990d07b2e21d971f2e6/src/Illuminate/Foundation/helpers.php#L229)
- [@return mixed|\Illuminate\Config\Repository](https://github.com/laravel/framework/blob/ab1506091b9f166b312b3990d07b2e21d971f2e6/src/Illuminate/Foundation/helpers.php#L263)
- [@return mixed|\Illuminate\Session\Store|\Illuminate\Session\SessionManager](https://github.com/laravel/framework/blob/ab1506091b9f166b312b3990d07b2e21d971f2e6/src/Illuminate/Foundation/helpers.php#L778)

Reply:
All those places are wrong, too.

Reply:
But isn't the purpose of PHPDoc to help the IDE do the autocomplete?
Because here `mixed` is not a type from PHP 8.0 and `barryvdh/laravel-ide-helper` doesn't handle such global functions.

Reply:
You're saying "everything but also this" which doesn't makes sense. I get the autocompletion part but from a documentation perspective this isn't correct.

Reply:
> barryvdh/laravel-ide-helper doesn't handle such global functions.

I think you can request a feature then on that library.

Reply:
Okay, I understand, it's completely wrong to write such hints on the one hand.

But there is no real standard: "You should never combine `mixed` with anything else in PHPDoc". This type hint is used even in the one of the [PSR packages](https://github.com/php-fig/http-message/blob/efd67d1dc14a7ef4fc4e518e7dee91c271d524e4/src/StreamInterface.php#L153).

In any case, are you okay with just reverting the changes without adding the `mixed` one? Because `request()` is one of the most commonly used helpers, and I don't want to lose autocomplete out of the box.

Reply:
Hey @stasadev. Let's let @taylorotwell decide on that.